armi/utils/reportPlotting.py

```diff
@@ -266,48 +266,28 @@ def buVsTime(reactor, scalars, extension=None):
     report.setData("Burnup Plot", os.path.abspath(figName), report.BURNUP_PLOT)
 
 
-def distortionVsTime(reactor, scalars, extension=None):
-    r"""plots max distortion vs. time if the distortion interface is attached"""
-    dd = reactor.o.getInterface("ductDistortion")
-    if not dd or not "maxSwelling" in dd.__dict__:
-        return  # skip plotting
-
-    extension = extension or settings.Settings()["outputFileExtension"]
-
-    plt.figure()
-    plt.plot(scalars["time"], dd.maxTotal, label="Total")
-    plt.plot(scalars["time"], dd.maxCreep, label="Creep")
-    plt.plot(scalars["time"], dd.maxSwelling, label="Swelling")
-    plt.xlabel("Time (yr)")
-    plt.ylabel("Distortion (mm)")
-    plt.grid(color="0.70")
-    plt.legend(loc="lower right")
-    plt.title("Maximum duct distortion for " + reactor.name)
-    figName = reactor.name + ".duct." + extension
-    plt.savefig(figName)
-    plt.close(1)
-
-    report.setData("Distortion Plot", os.path.abspath(figName), report.DISTORTION_PLOT)
-
-
-def xsHistoryVsTime(reactor, buGroups, extension=None):
+def xsHistoryVsTime(reactor, history, buGroups, extension=None):
     r"""
     Plot cross section history vs. time.
 
     Parameters
     ----------
     reactor : armi.reactor.reactors object
+
+    history : armi.bookkeeping.historyTracker.HistoryTrackerInterface object
+        The history interface.
+
     buGroups : list of float
         The burnup groups in the problem
+
     extension : str, optional
         The file extention for saving the figure
-
     """
     extension = extension or settings.Settings()["outputFileExtension"]
 
-    history = reactor.o.getInterface("history")
-    if not history or not history.xsHistory:
+    if nothistory.xsHistory:
```

Comment:
`not history`?

Reply:
OMG.

To be honest, one of the major reasons I like unit tests here: they catch my stupid typos.  Report plotting has really poor code coverage, currently, because it is excessively hard to test.  I will have to plan out some way of improving that.